shopinvader_sale_coupon/tests/test_cart.py

```diff
@@ -0,0 +1,156 @@
+# Copyright 2021 Camptocamp SA
+# @author Iván Todorovich <ivan.todorovich@gmail.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo.exceptions import UserError
+
+from odoo.addons.sale_coupon.tests.common import TestSaleCouponCommon
+from odoo.addons.shopinvader.tests.test_cart import CommonConnectedCartCase
+
+
+class TestCart(CommonConnectedCartCase, TestSaleCouponCommon):
+    def setUp(self, *args, **kwargs):
+        super().setUp(*args, **kwargs)
+        # Start with an empty cart
+        self.cart.order_line.unlink()
```

Comment:
can't you do this on setUpClass then?

Reply:
The thing is `cart` is set in `setUp` in `shopinvader` :disappointed: 

https://github.com/shopinvader/odoo-shopinvader/blob/fea64f5fdd6c2aa7644abd0018ff8285844f473a/shopinvader/tests/test_cart.py#L272-L287

It should probably be improved there, but such a change would impact other modules' tests
Alternatively:
- I can choose not to inherit from it, but that's kind of the whole point of that class
- I can probably do `self.env.ref("shopinvader.sale_order_2").order_line.unlink()` on `setUpClass`. Even though the cart is not referenced yet, it would have the same result I believe.. however if someone changes the cart referenced in `CommonConnectedCartCase` in the future, it would break this tests :/

What do you think?

Reply:
bah, ok, leave it like this and add a TODO pls :)